### PR TITLE
Fix Account Schema issues and ensure Admin account exists on start-up

### LINKED ISSUE #53: Ensure Admin account exists on start-up
(This is probably not the ideal way to do this, but it is a good compromise in order to get things rolling quickly)

Requires #52 

When loading the connection to the `mongo` service, we should check if an Admin account already exists. If not, we should create it based on environment values (`ADMIN_EMAIL` and `ADMIN_PASSWORD`).

### PR DESCRIPTION
**Pull request by miguelpduarte**
Closes #53 and addresses some spaghetti left over from #52

### CODE REVIEW

File: src/loaders/mongoose.js [hunk 1]
```diff
@@ -34,4 +36,24 @@ const setupDbConnection = async () => {
...
+const createDefaultAdmin = async () => {
+    if (process.env.NODE_ENV === "test") {
+        return;
+    }
+
+    const acc = await Account.findOne({ email: config.admin_email });
+    if (acc) {
+        // User with admin email already exists
+        return;
+    }
```
Comment (miguelpduarte) [inline src/loaders/mongoose.js]: Should we instead check for an (any) admin user already existing?

Comment (imnotteixeira) [reply src/loaders/mongoose.js]: I don't think so, we want THAT admin to exist, we don't want to find via trial and error which admin existed before

File: src/models/Account.js [hunk 2]
```diff
@@ -8,17 +8,15 @@ const AccountSchema = new Schema({
...
         type: Boolean,
-        required: function() {
-            return !this.companyRef;
-        },
+        default: false,
         validate: {
             validator: function(isAdmin) {
-                return isAdmin && !this.company;
+                return isAdmin !== !!this.company;
             },
```
Comment (imnotteixeira) [inline src/models/Account.js]: Can we swap to arrow function and make this a one-liner?

```js
validator: isAdmin => isAdmin !== !!this.company;
```

File: src/models/Account.js [hunk 3]
```diff
@@ -29,10 +27,10 @@ const AccountSchema = new Schema({
             return !this.isAdmin;
         },
         validate: {
-            validator: function(companyRef) {
-                return !!companyRef && !this.isAdmin;
+            validator: function(company) {
+                return !!company !== this.isAdmin;
             },
```
Comment (imnotteixeira) [inline src/models/Account.js]: Can we swap to arrow function and make this a one-liner?

```js
validator: company => !!company !== this.isAdmin;
```

File: src/models/Account.js (hunk 2 above)
Comment (miguelpduarte) [reply src/models/Account.js]: Arrow functions do not capture this, hence why the validators use the older `function` syntax :sweat_smile:

File: src/models/Account.js (hunk 3 above)
Comment (miguelpduarte) [reply src/models/Account.js]: See the other comment

File: src/loaders/mongoose.js (hunk 1 above)
Comment (miguelpduarte) [reply src/loaders/mongoose.js]: Ok, it was just checking for an admin with the given email vs checking for at least one admin being in the system. But this is fine for me as well. Resolving this thread then :slightly_smiling_face:

File: src/models/Account.js (hunk 2 above)
Comment (imnotteixeira) [reply src/models/Account.js]: Correct.. you would have to `.bind(this)`... nevermind

**Comment by imnotteixeira** [comment]
Merge this when you think best, don't know if it should go now or after #61

**Comment by miguelpduarte** [comment]
@imnotteixeira I'm waiting for #61 to merge first since it has more tests that might check for potential problems with conflicts here. But will merge it shortly after it's done.

**Comment by miguelpduarte** [comment]
Fixed conflicts, please check this and merge it if it's okay, I'll be back later if this still needs work, but I think it's probably done :slightly_smiling_face:

**Pull request merged**